src/traffic_cop/traffic_cop.cpp

```diff
@@ -275,6 +278,43 @@ TrafficCopResult TrafficCop::ExecuteDropStatement(
                                                common::ErrorCode::ERRCODE_DATA_EXCEPTION)};
 }
 
+TrafficCopResult TrafficCop::ExecuteExplain(const common::ManagedPointer<network::ConnectionContext> connection_ctx,
+                                            const common::ManagedPointer<network::PostgresPacketWriter> out,
+                                            const common::ManagedPointer<planner::AbstractPlanNode> physical_plan,
+                                            const terrier::network::QueryType query_type) const {
+  TERRIER_ASSERT(connection_ctx->TransactionState() == network::NetworkTransactionStateType::BLOCK,
```

Comment:
Let's refactor the body down to:
```
  TERRIER_ASSERT(connection_ctx->TransactionState() == network::NetworkTransactionStateType::BLOCK,
                 "Not in a valid txn. This should have been caught before calling this function.");
  // Need a single column to write back to the client
  std::vector<planner::OutputSchema::Column> output_columns;
  output_columns.emplace_back("QUERY PLAN", type::TypeId::VARCHAR, nullptr);
  out->WriteRowDescription(output_columns, {network::FieldFormat::text});

  // Dump to JSON string, wrap in StringVal, write the data row to the client
  const std::string plan_string = physical_plan->ToJson().dump(4);
  const execution::sql::StringVal plan_string_val =
      execution::sql::StringVal(plan_string.c_str(), plan_string.length());
  out->WriteDataRow(reinterpret_cast<const byte *const>(&plan_string_val), output_columns,
                    {network::FieldFormat::text});

  return {ResultType::COMPLETE, 0};
```
This cuts down on dummy variables, eliminates the switch with a single case, omits the `query_type` check that doesn't buy us much, and clarifies the logic.

Reply:
Note that passing the expression in as `nullptr` requires us to change postgres_packet_writer.cpp::78 to
```
    const auto &name = columns[i].GetExpr() == nullptr || columns[i].GetExpr()->GetAlias().empty()
                           ? columns[i].GetName()
                           : columns[i].GetExpr()->GetAlias();
```
To handle that expression being `nullptr`. To be honest, I don't understand why we have expressions in the output columns at all.